@jonahwilliams Here's the output:

```
$ flutter build apk --target-platform android-arm 
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Circular dependency between the following tasks:                        
:camera:generateReleaseRFile                                            
+--- :camera:generateReleaseRFile (*)                                   
\--- :path_provider:generateReleaseRFile                                
     +--- :camera:generateReleaseRFile (*)                              
     \--- :path_provider:generateReleaseRFile (*)                       
                                                                        
(*) - details omitted (listed previously)                               
                                                                        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 828ms                                                   
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                       1.4s
Gradle task assembleRelease failed with exit code 1
```

--- a/packages/flutter_tools/lib/src/plugins.dart
+++ b/packages/flutter_tools/lib/src/plugins.dart
@@ -281,7 +281,7 @@ Plugin _pluginFromPubspec(String name, Uri packageRoot) {
     name,
     packageRootPath,
     flutterConfig['plugin'] as YamlMap,
-    dependencies == null ? <String>[] : dependencies.keys.toList(),
+    dependencies == null ? <String>[] : <String>[...dependencies.keys],
   );
 }
 

--- a/packages/flutter_tools/lib/src/plugins.dart
+++ b/packages/flutter_tools/lib/src/plugins.dart
@@ -281,7 +281,7 @@ Plugin _pluginFromPubspec(String name, Uri packageRoot) {
     name,
     packageRootPath,
     flutterConfig['plugin'] as YamlMap,
-    dependencies == null ? <String>[] : <String>[...dependencies?.keys],
+    dependencies == null ? <String>[] : dependencies.keys.toList(),
   );
 }
 

--- a/packages/flutter_tools/lib/src/plugins.dart
+++ b/packages/flutter_tools/lib/src/plugins.dart
@@ -312,7 +312,7 @@ List<Plugin> findPlugins(FlutterProject project) {
 /// If there aren't any plugins, then the files aren't written to disk.
 ///
 /// Finally, returns [true] if .flutter-plugins or .flutter-plugins-dependencies have changed,
-/// otherwise returns false.
+/// otherwise returns [false].
 bool _writeFlutterPluginsList(FlutterProject project, List<Plugin> plugins) {
   final StringBuffer dependenciesFileBuffer = StringBuffer();
   final StringBuffer flutterPluginsBuffer = StringBuffer();

--- a/packages/flutter_tools/test/general.shard/plugins_test.dart
+++ b/packages/flutter_tools/test/general.shard/plugins_test.dart
@@ -262,7 +262,7 @@ dependencies:
         ProcessManager: () => FakeProcessManager.any(),
       });
 
-      testUsingContext('Refreshing the plugin list creates a plugin directory when there are plugins', () {
+      testUsingContext('Refreshing the plugin list modifies .flutter-plugins and .flutter-plugins-dependencies when there are plugins', () {
         createPluginWithDependencies(name: 'plugin-a', dependencies: const <String>['plugin-b', 'plugin-c', 'random-package']);
         createPluginWithDependencies(name: 'plugin-b', dependencies: const <String>['plugin-c']);
         createPluginWithDependencies(name: 'plugin-c', dependencies: const <String>[]);

--- a/packages/flutter_tools/test/general.shard/plugins_test.dart
+++ b/packages/flutter_tools/test/general.shard/plugins_test.dart
@@ -221,7 +221,6 @@ dependencies:
         );
     }
 
-
     // Creates the files that would indicate that pod install has run for the
     // given project.
     void simulatePodInstallRun(XcodeBasedProject project) {

